angular-client/src/components/battery/battery.component.ts

```diff
@@ -0,0 +1,55 @@
+import { Component, Input } from '@angular/core';
+import Theme from 'src/services/theme.service';
+
+@Component({
+  selector: 'battery',
+  templateUrl: 'battery.component.html',
+  styleUrls: ['battery.component.css']
+})
+export class BatteryPercentageComponent {
+  @Input() props!: { percentage: number; height: number; width: number };
+  color: string = '';
+  item!: {
+    full: boolean;
+    fillColor: string;
+    backColor: string;
+    height: string;
+    width: string;
+    batHeight: string;
+    nubHeight: string;
+    nubWidth: string;
+    roundCorner: string;
+  };
+
+  // setting color and rendering
+  ngOnInit() {
+    if (this.props.percentage <= 20) {
+      this.color = Theme.battteryLow;
+    } else if (this.props.percentage <= 50) {
+      this.color = Theme.battteryMed;
+    } else {
+      this.color = Theme.battteryHigh;
+    }
+    this.renderBattery();
+  }
+
+  // fills battery bar based on current percentage
+  // note:
+  renderBattery() {
+    const batHeight = ((100 - this.props.percentage) / 100) * this.props.height + 'px';
```

Comment:
does this work on different sized batteries?

Reply:
yessirski